exporters/stdout/stdoutlog/exporter.go

```diff
@@ -0,0 +1,87 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package stdoutlog // import "go.opentelemetry.io/otel/exporters/stdout/stdoutlog"
+
+import (
+	"context"
+	"encoding/json"
+	"sync/atomic"
+	"time"
+
+	"go.opentelemetry.io/otel/sdk/log"
+)
+
+var zeroTime time.Time
+
+var _ log.Exporter = &Exporter{}
+
+// Exporter is an implementation of  that writes spans to stdout.
+type Exporter struct {
+	encoder    *json.Encoder
+	timestamps bool
+
+	stopped atomic.Bool
+}
+
+// New creates an Exporter with the passed options.
+func New(options ...Option) (*Exporter, error) {
+	cfg := newConfig(options)
+
+	enc := json.NewEncoder(cfg.Writer)
+	if cfg.PrettyPrint {
+		enc.SetIndent("", "\t")
+	}
+
+	return &Exporter{
+		encoder:    enc,
+		timestamps: cfg.Timestamps,
+	}, nil
+}
+
+// Export exports log records to writer.
+// The writer is os.Stdout by default.
+func (e *Exporter) Export(ctx context.Context, records []log.Record) error {
+	if e.stopped.Load() {
+		return nil
+	}
+
+	for i, record := range records {
+		// Honor context cancellation.
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+
+		// Remove timestamps.
+		if !e.timestamps {
+			// Clone before make changes.
+			record = records[i].Clone()
+
+			record.SetTimestamp(zeroTime)
+			record.SetObservedTimestamp(zeroTime)
+		}
+
+		// Encode record, one by one.
+		recordJSON := newRecordJSON(record)
+		if err := e.encoder.Encode(recordJSON); err != nil {
```

Comment:
This will panic when a exporter is created with `&Exporter{}` instead of `New`.

Reply:
I am not sure we should expect the method to work without using the `New` function.

Reply:
See https://github.com/open-telemetry/opentelemetry-go/pull/5080#discussion_r1526739344 (and my response below).

If we decide to NOT have a usable zero value then we should at clearly say in `Exporter` GoDoc: `Exporter must be created with [New].`.

Reply:
Maybe we can un-export the `Exporter` so our users can only use the `New` function.

Reply:
I suggest having it exported as it allows adding functionalities to `Exporter` if needed. 

Similar issue https://github.com/open-telemetry/opentelemetry-go/issues/4247 when we were working on Metrics SDK.